### ISSUE TITLE
[CT-996] [Bug] commenting jinja template with using dummy package causes compilation error 

### ISSUE THREAD

**Issue by iamtodor** [dbt-labs/dbt-core#5616]
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

Is it a well-known bug, that when you have a commented (as usual within sql by -- ) jinja template in a select statement referencing some dummy uninstalled (inexistent) package your model wouldn't be compiled?
```
SELECT
    column_a
--    {{ dummy.some_func('column_b') }} as updated_column_b
FROM
    TableA
```
The error during the compilation:
```
Encountered an error:
Runtime Error
  Compilation Error in model my_model (path/to/my/model/my_model.sql)
    'dummy' is undefined. This can happen when calling a macro that does not exist. Check for typos and/or install package dependencies with "dbt deps".
```

I also tried the following - commenting jinja template in jinja style:
```
SELECT
    column_a
--    {{# dummy.some_func('column_b') #}} as updated_column_b
FROM
    TableA
```
and 
```
SELECT
    column_a
    {{# dummy.some_func('column_b') #}} as updated_column_b
FROM
    TableA
```

### Expected Behavior

I would expect anything that is being commented by two dashes should be ignored and shouldn't cause any compilation errors

### Steps To Reproduce

```
SELECT
    column_a
--    {{ dummy.some_func('column_b') }} as updated_column_b
FROM
    TableA
```

### Relevant log output

```shell
Encountered an error:
Runtime Error
  Compilation Error in model my_model (path/to/my/model/my_model.sql)
    'dummy' is undefined. This can happen when calling a macro that does not exist. Check for typos and/or install package dependencies with "dbt deps".
```

### Environment

- OS: 
```
>>> cat /etc/*-release
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
>>> airflow@4c6a8d1e7037:~/dbt-tasks/packages$ cat /proc/version
Linux version 5.10.104-linuxkit (root@buildkitsandbox) (gcc (Alpine 10.2.1_pre1) 10.2.1 20201203, GNU ld (GNU Binutils) 2.35.2) #1 SMP Thu Mar 17 17:08:06 UTC 2022
```

- Python: 3.8.12
- dbt:
```
installed version: 0.20.1
   latest version: 1.0.0

Plugins:
  - bigquery: 0.20.1
```

### Which database adapter are you using with dbt?

bigquery

### Additional Context

It might relate to https://github.com/dbt-labs/dbt-core/issues/5305 but I wouldn't say it's exactly the same

**Comment by lostmygithubaccount** [comment]
Hi @iamtodor, thanks for reporting this! I'm going to close as there's no engineering action to take here, this is known and "intended" behavior. It did also trip me up as I was recently onboarding to dbt Labs for what that's worth :) as you noted https://github.com/dbt-labs/dbt-core/issues/5305 has some similar behavior, as does https://github.com/dbt-labs/dbt-core/issues/458 discussed amongst our co-founders long ago.

Your most immediate issue, I believe, is using `{{# ... #}}` as your commenting syntax -- it should be `{# ... #}`, see screenshots below.

Attempting to use `--` will result in bad compilation:

![Image](https://user-images.githubusercontent.com/54814569/186029951-8db6ff88-8f00-4197-ab35-197e5a22d1cc.png)

Basically the macro still gets injected and only the first line is commented. 

Using `{# ... #}` works as desired, with or without `--`:

![Image](https://user-images.githubusercontent.com/54814569/186030185-e5ba8e06-eac1-4ba5-916f-0830a22a43a4.png)

![Image](https://user-images.githubusercontent.com/54814569/186030250-8b54c35f-1221-42af-bb02-5ddf0539e3c4.png)

Attempting to use `{{# ... #}}` won't compile: 

![Image](https://user-images.githubusercontent.com/54814569/186030084-dd25a082-aaca-4284-84e4-59bd82787edc.png)
